### PR TITLE
go/pup: init at 0.3.9

### PR DESCRIPTION
**Pull request by obadz**
(also required go/color: init at 9aae6aa)

First time I package a Go package.
- What's the `date` field for?
- Is there a better way to do what I did in `renameImports`?
- Am I meant to invoke this as `goPackages.pup.bin`?

### CODE REVIEW

**Comment by obadz** [comment]
cc @jb55

**Comment by obadz** [comment]
Throwing in go-mtpfs for good measure.

Still don't know what the `date` field is for.

File: pkgs/top-level/go-packages.nix [hunk 1]
```diff
@@ -3182,4 +3182,29 @@ let
...
+
+  pup = buildFromGitHub {
+    rev      = "9693b292601dd24dab3c04bc628f9ae3fa72f831";
+    date     = "2015-11-27";
+    owner    = "EricChiang";
+    repo     = "pup";
+    sha256   = "04j3fy1vk6xap8ad7k3c05h9b5mg2n1vy9vcyg9rs02cb13d3sy0";
+    propagatedBuildInputs = [ net ];
+    buildInputs = [ go-colorable color ];
+    renameImports = ''
```
Comment (lethalman) [inline pkgs/top-level/go-packages.nix]: Do it in a `preBuild` or `postPatch` or somewhere else. `renameImports` is for overriding the default renaming.

**Comment by lethalman** [comment]
About `date`, there's the source code for `buildFromGitHub`. Should be quite clear.

About `.bin`, yes.

**Comment by obadz** [comment]
fixed as requested.

**Comment by obadz** [comment]
@lethalman, anything else before this can get merged?

**Pull request merged**